homeassistant/components/sensor/buienradar.py

```diff
@@ -0,0 +1,293 @@
+"""
+Support for Buienradar.nl weather service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.buienradar/
+"""
+import asyncio
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_LATITUDE, CONF_LONGITUDE, CONF_MONITORED_CONDITIONS,
+    ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import (
+    async_track_point_in_utc_time)
+from homeassistant.util import dt as dt_util
+
+REQUIREMENTS = ['buienradar==0.4', 'xmltodict==0.11.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+# Sensor types are defined like so:
+# SENSOR_TYPES = { 'key': ['Display name',
+#                          'unit of measurement',
+#                          icon (or None),
+#                }
+SENSOR_TYPES = {
+    'stationname': ['Stationname', None, None],
+    'symbol': ['Symbol', None, None],
+    'humidity': ['Humidity', '%', 'mdi:water-percent'],
+    'temperature': ['Temperature', '°C', 'mdi:thermometer'],
+    'groundtemperature': ['Ground Temperature', '°C', 'mdi:thermometer'],
+    'windspeed': ['Wind speed', 'm/s', 'mdi:weather-windy'],
+    'windforce': ['Wind force', 'Bft', 'mdi:weather-windy'],
+    'winddirection': ['Wind direction', '°', 'mdi:compass-outline'],
+    'windazimuth': ['Wind direction azimuth', None, 'mdi:compass-outline'],
+    'pressure': ['Pressure', 'hPa', 'mdi:gauge'],
+    'visibility': ['Visibility', 'm', None],
+    'windgust': ['Wind gust', 'm/s', 'mdi:weather-windy'],
+    'precipitation': ['Precipitation', 'mm/h', 'mdi:weather-pouring'],
+    'irradiance': ['Irradiance', 'W/m2', 'mdi:sunglasses'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_MONITORED_CONDITIONS,
+                 default=['symbol', 'temperature']): vol.All(
+                     cv.ensure_list, vol.Length(min=1),
+                     [vol.In(SENSOR_TYPES.keys())]),
+    vol.Optional(CONF_LATITUDE): cv.latitude,
+    vol.Optional(CONF_LONGITUDE): cv.longitude,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup the buienradar_nl sensor."""
```

Comment:
Can you disclose why you think it will not pass? I am happy to make any adjustments here...
Just to check, I ran pep257, which passed both files...